media/src/main/java/com/example/media/camera/CameraSnippets.kt

```diff
@@ -0,0 +1,89 @@
+package com.example.media.camera
+
+import android.content.Context
+import androidx.camera.compose.CameraXViewfinder
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageCapture
+import androidx.camera.core.Preview
+import androidx.camera.core.SurfaceRequest
+import androidx.camera.core.UseCaseGroup
+import androidx.camera.core.takePicture
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.camera.lifecycle.awaitInstance
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.compose.LifecycleStartEffect
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+class CameraPreviewViewModel(private val appContext: Context): ViewModel() {
+    private val _surfaceRequest = MutableStateFlow<SurfaceRequest?>(null)
+    val surfaceRequest: StateFlow<SurfaceRequest?> = _surfaceRequest
+
+    private lateinit var processCameraProvider: ProcessCameraProvider
+    private val previewUseCase = Preview.Builder().build()
+    private val captureUseCase = ImageCapture.Builder().build()
+    private val useCaseGroup = UseCaseGroup.Builder().apply {
+        addUseCase(previewUseCase)
+        addUseCase(captureUseCase)
+    }.build()
+
+    private var runningCameraJob: Job? = null
+
+    fun startCamera() {
+        stopCamera()
+        runningCameraJob = viewModelScope.launch {
+            processCameraProvider = ProcessCameraProvider.awaitInstance(appContext)
+            processCameraProvider.runWith(CameraSelector.DEFAULT_BACK_CAMERA, useCaseGroup) {
+                previewUseCase.setSurfaceProvider { newSurfaceRequest ->
+                    _surfaceRequest.value = newSurfaceRequest
+                }
+            }
+        }
+    }
+
+    fun stopCamera() {
+        runningCameraJob?.apply {
+            if (isActive) {
+                cancel()
+            }
+        }
+    }
+
+    fun takePicture() {
+        viewModelScope.launch {
+            val imageProxy = captureUseCase.takePicture()
+            // Do something with the image
+        }
+    }
+}
+
+@Composable
+fun CameraPreviewScreen(modifier: Modifier = Modifier) {
+    val viewModel = CameraPreviewViewModel(LocalContext.current.applicationContext)
```

Comment:
ViewModel should be retrieved via `androidx.lifecycle.viewmodel.compose.viewModel`. Something like:
```
@Composable
fun CameraPreviewScreen(
    modifier: Modifier = Modifier,
    viewModel: CameraPreviewViewModel = viewModel()
) {
    val surfaceRequest by viewModel.surfaceRequest.collectAsStateWithLifecycle()
```